### ISSUE TITLE
Fragile API endpoint when creating a new flag

### ISSUE THREAD

**Issue by avishail** [SolidStateGroup/bullet-train-api#1818]
I was trying to use your API in order to create a flag.

URL that won't work: https://api.flagsmith.com/api/v1/projects/<PROJECT_ID>/features
URL that will work: https://api.flagsmith.com/api/v1/projects/<PROJECT_ID>/features/

the trailing / will decide whether the POST request will actually create a flag or the whole request will be treated as a "query" and the list of my flags will be returned.

**Comment by matthewelwell** [comment]
Hi @avishail,

Thanks for getting in touch. 

The POST endpoint specifically requires the trailing slash, the URL without the trailing slash not a valid endpoint - this is the main reason why you're seeing this behaviour. You can see from the following screenshot of the swagger documentation that the endpoint is listed with the trailing slash. 

<img width="1474" alt="image" src="https://user-images.githubusercontent.com/14089968/210378605-af462f26-b002-4fba-a62d-2e91f5dfdfa2.png">

What is perhaps adding some confusion is that, in an effort to help in certain scenarios, when a matching route is not found for a request that does not include a trailing slash, it will attempt to redirect the client to the same URL with the trailing slash appended (you can read about this behaviour a little more [here](https://docs.djangoproject.com/en/4.1/ref/settings/#append-slash) if you're interested). The behaviour you're seeing is that your client is following that redirect (but changing the method to GET). This is the default behaviour in cURL for example, however, it can be changed by setting the `-X` option as seen in the example below. 

```
curl 'https://api.flagsmith.com/api/v1/projects/<project id>/features' 
  -X 'POST' \
  -H 'authorization: Token <token>' \
  -H 'content-type: application/json; charset=utf-8' \
  --data-raw '{"name":"test","tags":[],"project":<project id>,"type":"STANDARD"}' \
  --compressed
  --location
```

This cURL request will now give you a 400 error since it has tried to redirect the original POST request to the correct URL (with the trailing slash), however it cannot do so while maintaining the request body, and hence it throws a validation error. 

Hopefully this helps to explain a little about what is happening.

tl;dr - The endpoint without the trailing slash is incorrect. It must have the trailing slash. I'm going to go ahead and close this issue on this basis.

**Comment by avishail** [comment]
Thanks for the detailed answer.
